src/tests/index.test.ts

```diff
@@ -87,25 +99,19 @@ it( 'should await the app ready state before creating a window for activate even
 				},
 			};
 		} );
-		const createMainWindowMock = jest.fn();
-		jest.doMock( '../main-window', () => {
-			return {
-				createMainWindow: createMainWindowMock,
-			};
-		} );
 		require( '../index' );
 
 		activate();
 
-		expect( createMainWindowMock ).not.toHaveBeenCalled();
+		expect( createMainWindow ).not.toHaveBeenCalled();
 		// Await the mocked `whenReady` promise resolution
 		await new Promise( process.nextTick );
-		expect( createMainWindowMock ).toHaveBeenCalled();
+		expect( createMainWindow ).toHaveBeenCalled();
 	} );
 } );
 
-it( 'should gracefully handle app ready failures when creating a window on activate', () => {
-	jest.isolateModules( async () => {
+it( 'should gracefully handle app ready failures when creating a window on activate', async () => {
+	await jest.isolateModulesAsync( async () => {
```

Comment:
Following [Jest documentation](https://jestjs.io/docs/jest-object#jestisolatemodulesasyncfn), we should use the async version of `isolateModules` when there's async logic involved.